### PR TITLE
fix: multiplexed session metrics were not included in refactor move

### PR DESCRIPTION
**Pull request by olavloite**
The metrics for multiplexed sessions were not included in the refactoring that moved multiplexed sessions out of the session pool. This change re-adds those metrics based on the new client for multiplexed sessions.

### CODE REVIEW

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java [hunk 1]
```diff
@@ -3143,7 +3153,7 @@ boolean isValid() {
 
   /**
    * Returns a multiplexed session. The method fallbacks to a regular session if {@link
-   * SessionPoolOptions#useMultiplexedSession} is not set.
+   * SessionPoolOptions#getUseMultiplexedSession} is not set.
```
Comment (olavloite) [inline google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java:3156]: This reference was wrong, and showed up as a warning in IntelliJ

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java [hunk 2]
```diff
@@ -68,7 +68,6 @@
 import com.google.common.base.Function;
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Preconditions;
-import com.google.common.base.Supplier;
```
Comment (olavloite) [inline google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java:71]: This is not directly related to this change, but we should start to replace the use of `com.google.common.base.Supplier` with `java.util.function.Supplier`.

**Review by rahul2393 — APPROVED** [review]
